src/main/java/org/nixos/idea/psi/impl/NixParserUtil.java

```diff
@@ -36,4 +37,15 @@ public static boolean parseIsBindValue(@NotNull PsiBuilder builder, int level) {
     Boolean isInBinding = builder.getUserData(IS_IN_BINDING);
     return isInBinding != null && isInBinding;
   }
+
+  /**
+   * Calls "next" only if it's not occurring after a whitespace token.
+   * Whitespace is automatically skipped by {@link PsiBuilder}, therefore we're checking the previous raw token
+   * and the current raw token.
+   */
+  public static boolean not_after_whitespace(@NotNull PsiBuilder builder, int level, @NotNull Parser next) {
+    return builder.rawLookup(-1) != TokenType.WHITE_SPACE
+            && builder.rawLookup(0) != TokenType.WHITE_SPACE
+            && next.parse(builder, level);
+  }
```

Comment:
I am fine with this solution. Just out of curiosity, is there a specific reason for not using an zero-length `PATH_END` token?

Reply:
Sure, that's possible, too. This was most out of personal taste and previous experience with the raw token handling.
I've updated this with d82beb0b7c9b52609b67997a59ac295cac72b077

Reply:
As I have said, I would be fine with your previous solution as well. I can see that you had to add a special case in `NixSyntaxHighlighterTest` for the `PATH_END` token, which is also not ideal, I guess. Anyway, I am still fine with both solutions. If you want to keep the `PATH_END` token, you should probably add it to `NixTypeUtil.MIGHT_COLLAPSE_WITH_ID`.

Reply:
IMHO using `PATH_END` is slightly better and it certainly simplifies the parser definition.
425bfd3692d6231192591840bce958a7d4a8a6a7 updates the parser definition.